foreign_cc/nmake.bzl

```diff
@@ -0,0 +1,57 @@
+"""Helper rules and macros for building projects using the MSVC nmake build tool"""
+
+load("//foreign_cc:providers.bzl", "ForeignCcDepsInfo")
+load(":configure.bzl", "configure_make")
+
+def _extra_toolchains_transition_impl(settings, attrs):
+    return {"//command_line_option:extra_toolchains": attrs.extra_toolchains + settings["//command_line_option:extra_toolchains"]}
```

Comment:
I'm not too familiar with extra toolchains, can you explain how this can use a toolchain that isn't registered? What's the guarantee that the thing that's returned here is going to be used for the target if another toolchain was matched?

Reply:
Sure

[`extra_toolchains`](https://docs.bazel.build/versions/main/command-line-reference.html#flag--extra_toolchains) are considered before those registered in a WORKSPACE file (by `register_toolchains`). As such, by applying the transition it is like the target is being built like `bazel build <target> --extra_toolchains="@rules_foreign_cc//toolchains:preinstalled_nmake_toolchain"` and the nmake toolchain will be considered before the toolchains registered in the WORKSPACE file

Reply:
I personally prefer this approach to transitioning the host platform as I think it is less confusing.

When transitioning the host platform, the transition is from a Windows OS platform targeting MSVC to a Windows OS platform targeting MSVC. Essentially the approach leverages [this](https://github.com/bazelbuild/bazel/issues/7730) issue.